### PR TITLE
PMM-11079 Remove Integrated Alerting

### PR DESCRIPTION
**Pull request by artemgavrilov**
PMM-11079

Link to the Feature Build: SUBMODULES-3324

If this PR adds or removes or alters one or more API endpoints, please review and add or update the relevant [API documents](https://github.com/percona/pmm/tree/main/docs/api) as well:

- [ ] API Docs updated

If this PR is related to some other PRs in this or other repositories, please provide links to those PRs:

- Links to related pull requests (optional).

### CODE REVIEW

File: api/serverpb/server.proto [hunk 1]
```diff
@@ -351,17 +308,6 @@ service Server {
...
-  rpc TestEmailAlertingSettings(TestEmailAlertingSettingsRequest) returns (TestEmailAlertingSettingsResponse) {
-    option (google.api.http) = {
-      post: "/v1/Settings/TestEmailAlertingSettings"
-      body: "*"
-    };
-    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
-      summary: "Test email alerting"
-      description: "Sends test email to check current SMTP settings for email alerting."
-    };
-  }
```
Comment (artemgavrilov) [inline api/serverpb/server.proto:364]: It's breaking change, but it has no sense to keep this endpoint

Comment (atymchuk) [reply api/serverpb/server.proto:364]: Let's link a doc ticket to PMM-11079 so we mention it in RN.

**Comment by atymchuk** [comment]
I believe there could be more mentions of `Integrated Alerting` in the code, for example https://github.com/percona/pmm/blob/6e252ecdd7448483979f3622e941273556004268/managed/services/errors.go#L23

File: managed/services/management/alerting/service.go [hunk 2]
```diff
@@ -942,3 +987,36 @@ func validateParameters(definitions models.AlertExprParamsDefinitions, values mo
...
+type AlertExprParamsValues []AlertExprParamValue
+
+// AsStringMap convert param values to string map, where parameter name is a map key and parameter value is a map value.
+func (p AlertExprParamsValues) AsStringMap() map[string]string {
+	m := make(map[string]string, len(p))
+	for _, rp := range p {
+		var value string
+		switch rp.Type {
+		case models.Float:
+			value = fmt.Sprint(rp.FloatValue)
```
Comment (atymchuk) [inline managed/services/management/alerting/service.go:1010]: I wonder how can this pass the `forbidigo` linter rule, which has been [enabled](https://github.com/percona/pmm/pull/2302/files#diff-6179837f7df53a6f05c522b6b7bb566d484d5465d9894fb04910dd08bb40dcc9L118) not long ago? Is the linter failing to detect it here or is it misconfigured by any chance?

File: api-tests/management/alerting/alerting_test.go [hunk 3]
```diff
@@ -16,121 +16,46 @@
...
 
 	pmmapitests "github.com/percona/pmm/api-tests"
 	alertingClient "github.com/percona/pmm/api/managementpb/alerting/json/client"
 	"github.com/percona/pmm/api/managementpb/alerting/json/client/alerting"
+	"github.com/percona/pmm/managed/services/grafana"
 )
 
 // Note: Even though the IA services check for alerting enabled or disabled before returning results
 // we don't enable or disable IA explicit in our tests since it is enabled by default through
 // ENABLE_ALERTING env var.
```
Comment (atymchuk) [inline api-tests/management/alerting/alerting_test.go:44]: I understand ENABLE_ALERTING is no longer relevant, but it's counterpart DISABLE_ALERTING is.

Also, there are mentions of IA up above, which, AFAICS, get changed to Percona Alerting in this PR.

File: managed/services/server/deps.go [hunk 4]
```diff
@@ -116,18 +114,7 @@ type agentsStateUpdater interface {
...
-	WriteVMAlertRulesFiles()
-	RemoveVMAlertRulesFiles() error
-}
-
-type emailer interface {
-	Send(ctx context.Context, settings *models.EmailAlertingSettings, emailTo string) error
-}
-
-// rulesService is a subset of methods of ia.TemplatesService used by this package.
+// templatesService is a subset of methods of ia.TemplatesService used by this package.
```
Comment (idoqo) [inline managed/services/server/deps.go:117]:
```suggestion
// templatesService is a subset of methods of alerting.Service used by this package.
```

**Review by idoqo — APPROVED** [review]

File: managed/services/management/alerting/service.go (hunk 2 above)
Comment (idoqo) [reply managed/services/management/alerting/service.go:1010]: Isn't it meant to catch instances of `fmt.Print*` (i.e., print debugging) and not `fmt.Sprint*`?

Comment (atymchuk) [reply managed/services/management/alerting/service.go:1010]: Oh yes, that's the default configuration :) Thanks!

File: managed/services/server/deps.go (hunk 4 above)
Comment (artemgavrilov) [reply managed/services/server/deps.go]: Fixed

**Comment by artemgavrilov** [comment]
I removed `api/swagger/swagger-dev-only.json` because there is not target that updates it anymore.

File: api-tests/management/alerting/alerting_test.go (hunk 3 above)
Comment (artemgavrilov) [reply api-tests/management/alerting/alerting_test.go]: Cleaned up this, and all `IA`, `ia`, `integrated alerting` entries.

**Review by atymchuk — APPROVED** [review]

**Comment by artemgavrilov** [comment]
This PR is postponed until PMM v3

File: api/serverpb/server.proto [hunk 5]
```diff
@@ -146,6 +120,8 @@ message STTCheckIntervals {
 
 // Settings represents PMM Server settings.
 message Settings {
+  reserved 13, 14; // Removed fields
```
Comment (artemgavrilov) [inline api/serverpb/server.proto:123]: I guess we can just remove them now, and reorder all fields. Or that can be done in separate ticket for the whole API.

Comment (atymchuk) [reply api/serverpb/server.proto:123]: It's not the client, I also think we can go ahead and just delete.

Comment (artemgavrilov) [reply api/serverpb/server.proto:123]: Let's do this in the scope of whole API review

**Comment by artemgavrilov** [comment]
@BupycHuk @ademidoff  I think it's ready to be merged

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**